src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,127 @@
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+public class Exercise1 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        class PersonAgeComparator implements Comparator<Person> {
+
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                return firstPerson.getAge() - secondPerson.getAge();
+            }
+        }
+
+        Arrays.sort(persons, new PersonAgeComparator());
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                return firstPerson.getAge() - secondPerson.getAge();
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person firstPerson, Person secondPerson) {
+                int res = firstPerson.getLastName()
+                                     .compareTo(secondPerson.getLastName());
+                return res == 0 ? firstPerson.getFirstName()
+                                             .compareTo(secondPerson.getFirstName()) : res;
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+        Predicate<Person> personAgeEqualsThirtyPredicate = new Predicate<Person>() {
```

Comment:
Here should be your Guava Predicate implementation, not anonymous class

Reply:
Fixed here: bd894c6aee13f5f477b0bea17754593985f3c8fe